### PR TITLE
[coreSNTP] CBMC Proof for Sntp_SerializeRequest Function

### PR DESCRIPTION
**Pull request by gshvang**
This PR contains CBMC Proof for Sntp_SerializeRequest Function.

### CODE REVIEW

File: tools/uncrustify.cfg [hunk 1]
```diff
@@ -0,0 +1,160 @@
+# Uncrustify-0.67
```
Comment (aggarw13) [inline tools/uncrustify.cfg:1]: Has this file been accidentally added?

File: test/cbmc/proofs/Sntp_SerializeRequest/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+To run the proof.
+-------------
+
+* Add `cbmc`, `goto-cc`, `goto-instrument`, `goto-analyzer`, and `cbmc-viewer`
+  to your path.
+* Run `make`.
+* Open html/index.html in a web browser.
+
+To use [`arpa`](https://github.com/awslabs/aws-proof-build-assistant) to simplify writing Makefiles.
```
Comment (aggarw13) [inline test/cbmc/proofs/Sntp_SerializeRequest/README.md:17]: This sentence seems incomplete

File: test/cbmc/proofs/Sntp_SerializeRequest/Sntp_SerializeRequest_harness.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+void harness()
+{
+    SntpTimestamp_t * pRequestTime;
+    uint32_t randomNumber;
+    void * pBuffer;
+    size_t bufferSize;
+
+    pRequestTime = malloc( sizeof( SntpTimestamp_t ) );
+
+    pBuffer = malloc( bufferSize );
```
Comment (yourslab) [inline test/cbmc/proofs/Sntp_SerializeRequest/Sntp_SerializeRequest_harness.c:40]:
```suggestion
   __CPROVER_ASSUME( bufferSize < CBMC_MAX_OBJECT_SIZE );
    pBuffer = malloc( bufferSize );
```

**Review by yourslab — APPROVED** [review]

File: test/cbmc/proofs/Sntp_SerializeRequest/Sntp_SerializeRequest_harness.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    void * pBuffer;
+    size_t bufferSize;
+
+    pRequestTime = malloc( sizeof( SntpTimestamp_t ) );
+
+    __CPROVER_assume( bufferSize < CBMC_MAX_OBJECT_SIZE );
+
+    pBuffer = malloc( bufferSize );
+
+    Sntp_SerializeRequest( pRequestTime, randomNumber, pBuffer, bufferSize );
```
Comment (dan4thewin) [inline test/cbmc/proofs/Sntp_SerializeRequest/Sntp_SerializeRequest_harness.c:44]: Is there anything meaningful we can assert about the results of running Sntp_SerializeRequest()?

Comment (gshvang) [reply test/cbmc/proofs/Sntp_SerializeRequest/Sntp_SerializeRequest_harness.c:44]: I don't think as it is just serializing the request into the provided buffer and returns the SntpStatus_t.

Comment (yourslab) [reply test/cbmc/proofs/Sntp_SerializeRequest/Sntp_SerializeRequest_harness.c:44]: In that case, could `__CPROVER_assert` the return value and contents of `pRequestTime`.

Comment (gshvang) [reply test/cbmc/proofs/Sntp_SerializeRequest/Sntp_SerializeRequest_harness.c:44]: To achieve that , a separate function for checking the status value will do as the Sntp_SerializeRequest() can return different status values.

File: test/cbmc/proofs/Sntp_SerializeRequest/README.md (hunk 2 above)
Comment (gshvang) [reply test/cbmc/proofs/Sntp_SerializeRequest/README.md:17]: This is autogenerated text generated by proof set up script.

**Review by aggarg — APPROVED** [review]

**Pull request merged**